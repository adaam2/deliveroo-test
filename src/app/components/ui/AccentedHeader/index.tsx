import React from 'react';

import styles from "./AccentedHeader.module.scss";
import { Container } from 'ui';

export interface AccentedHeaderProps {
  children: any;
  className?: string;
}

const AccentedHeader: React.SFC<AccentedHeaderProps> = ({ children, className }) => {
  return (
    <header className={styles.accentedHeader}>
      <Container className={className}>
        {children}
      </Container>
    </header>
  );
}

export default AccentedHeader;