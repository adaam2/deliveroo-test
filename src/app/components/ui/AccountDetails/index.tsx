import React from 'react';

import { Icon, P } from '../';
import styles from "./AccountDetails.module.scss";

export interface AccountDetailsProps {
  userName: string;
}

const AccountDetails: React.SFC<AccountDetailsProps> = ({ userName }) => {
  return (
    <div className={styles.wrapper}>
      <div className={styles.flexItem}>
        <Icon iconKey="user" />
      </div>
      <div className={styles.userName}>
        <P marginLeft={8} fontSize={16} colour="#2e3333">
          {userName}
        </P>
      </div>
    </div>
  );
}

export default AccountDetails;