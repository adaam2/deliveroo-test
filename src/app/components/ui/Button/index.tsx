import React from 'react';
import classNames from "classnames/bind";

import styles from "./Button.module.scss";

const cx = classNames.bind(styles);

export enum ButtonType {
  Primary,
  Secondary
}

export interface ButtonProps {
  type: ButtonType,
  children: any;
  onClick(e: React.MouseEvent<HTMLButtonElement, MouseEvent>): void;
  ariaLabel?: string;
}

const Button: React.SFC<ButtonProps> = ({
  type = ButtonType.Primary,
  children,
  onClick,
  ariaLabel
}) => {
  const buttonClasses = cx({
    [styles.button]: true,
    [styles.primary]: type === ButtonType.Primary,
    [styles.secondary]: type === ButtonType.Secondary
  });

  return (
    <button aria-label={ariaLabel} onClick={onClick} className={buttonClasses}>
      {children}
    </button>
  );
}

export default Button;