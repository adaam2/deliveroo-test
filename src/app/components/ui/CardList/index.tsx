import React from 'react';

import styles from "./CardList.module.scss";

export interface CardListProps {
  children: any;
}

const CardList: React.SFC<CardListProps> = ({ children }) => {
  return (
    <div className={styles.list}>
      {children}
    </div>
  );
}

export default CardList;