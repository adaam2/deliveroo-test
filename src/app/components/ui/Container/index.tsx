import React from 'react';
import classNames from "classnames/bind";

import styles from "./Container.module.scss";

const cx = classNames.bind(styles);

export interface ContainerProps {
  children: any;
}

const Container: React.SFC<ContainerProps> = ({ children }) => {
  const containerClasses = () => cx({
    [styles.containerBase]: true
  });

  return (
    <div className={containerClasses()}>
      {children}
    </div>
  );
}

export default Container;