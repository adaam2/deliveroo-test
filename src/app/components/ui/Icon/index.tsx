import React from 'react';

export interface IconProps {
  altText?: string;
}

const Icon: React.SFC<IconProps> = ({ altText }) => {
  const icon = require(`assets/svgs/user.svg`);

  return (
    <>
      <img alt={altText} src={icon} />
    </>
  );
}

export default Icon;