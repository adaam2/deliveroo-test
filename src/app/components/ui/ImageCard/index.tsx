import React from 'react';

import styles from "./ImageCard.module.scss";

export interface ImageCardProps {
  imageUrl: string;
  children: any;
  altText: string;
}

const ImageCard: React.SFC<ImageCardProps> = ({ imageUrl, altText, children }) => {
  return (
    <div className={styles.card}>
      <img src={imageUrl} alt={altText} className={styles.img} />

      <div className={styles.cardContent}>
        {children}
      </div>
    </div>
  );
}

export default ImageCard;