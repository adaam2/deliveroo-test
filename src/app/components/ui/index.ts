import Nav from "./Nav";
import Logo from "./Logo";
import Icon from "./Icon";
import { P, Title } from "./Typography";
import ImageCard from "./ImageCard";
import Container from "./Container";
import Spacer from "./Spacer";

export { Nav, Logo, Icon, P, Title, ImageCard, Container, Spacer };
