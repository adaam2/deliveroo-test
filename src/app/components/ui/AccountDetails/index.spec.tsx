import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";

import { AccountDetails } from "ui";

let wrapper: ShallowWrapper;

function setup(
  userName: string
) {
  return shallow(
    <AccountDetails userName={userName} />
  );
};

describe('<AccountDetails />', () => {
  beforeEach(() => {
    wrapper = setup("Adam Bull");
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the users name', () => {
    expect(wrapper.html())
      .toEqual(expect.stringContaining('Adam Bull'));
  });

  it('renders the user icon', () => {
    expect(wrapper.html())
      .toEqual(expect.stringContaining('user.svg'));
  });
});
