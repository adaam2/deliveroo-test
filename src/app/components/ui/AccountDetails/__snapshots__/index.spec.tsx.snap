// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountDetails /> matches the snapshot 1`] = `
<div>
  <div>
    <Icon />
  </div>
  <div>
    <Component
      colour="#2e3333"
      fontSize={16}
      marginLeft={8}
    >
      Adam Bull
    </Component>
  </div>
</div>
`;
